source/server/http/admin.cc

```diff
@@ -520,17 +551,86 @@ Http::Code AdminImpl::runCallback(const std::string& path, Buffer::Instance& res
     for (auto handler : sorted_handlers) {
       response.add(fmt::format("  {}: {}\n", handler.first, handler.second->help_text_));
     }
+    code = Http::Code::NotFound;
   }
 
   return code;
 }
 
+Http::Code AdminImpl::handlerAdminHome(const std::string&, Http::HeaderMap& header_map,
+                                       Buffer::Instance& response) {
+  header_map.addReferenceKey(Http::Headers::get().ContentType, "text/html; charset=UTF-8");
+
+  // Prefix order is used during searching, but for printing do them in alpha order.
+  std::map<std::string, const UrlHandler*> sorted_handlers;
+  for (const UrlHandler& handler : handlers_) {
+    sorted_handlers[handler.prefix_] = &handler;
+  }
+
+  response.add(R"(
+<head>
+  <link rel='shortcut icon' type='image/png' href='https://www.envoyproxy.io/img/favicon.ico'/>
```

Comment:
This might break when accessing in environments without public net access.

Reply:
I thought of that and felt that the favicon failure in those situations would not be a big deal, but I like to have favicons in the tabs when available because sometimes that's all you can see.  
However, on reflection it is kind of unprofessional to have the 4xx response.

The favicon ico source is annoyingly big to include in the server,  so I wouldn't want to inline it.  But I found that if I screen-captured the favicon as a PNG and base64 encoded it, it's not too bad (846 bytes), so I went for that.